contracts/verifiers/Verifier.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interface/IClaimIssuer.sol";
+
+contract Verifier is Ownable {
+    /// @dev All topics of claims required to pass verification.
+    uint256[] public requiredClaimTopics;
+
+    /// @dev Array containing all TrustedIssuers identity contract address allowed to issue claims required.
+    IClaimIssuer[] public trustedIssuers;
+
+    /// @dev Mapping between a trusted issuer address and the topics of claims they are trusted for.
+    mapping(address => uint256[]) public trustedIssuerClaimTopics;
+
+    /// @dev Mapping between a claim topic and the trusted issuers trusted for it.
+    mapping(uint256 => IClaimIssuer[]) public claimTopicsToTrustedIssuers;
+
+    /**
+     *  this event is emitted when a claim topic has been added to the requirement list
+     *  the event is emitted by the 'addClaimTopic' function
+     *  `claimTopic` is the required claim topic added
+     */
+    event ClaimTopicAdded(uint256 indexed claimTopic);
+
+    /**
+     *  this event is emitted when a claim topic has been removed from the requirement list
+     *  the event is emitted by the 'removeClaimTopic' function
+     *  `claimTopic` is the required claim removed
+     */
+    event ClaimTopicRemoved(uint256 indexed claimTopic);
+
+    /**
+     *  this event is emitted when an issuer is added to the trusted list.
+     *  the event is emitted by the addTrustedIssuer function
+     *  `trustedIssuer` is the address of the trusted issuer's ClaimIssuer contract
+     *  `claimTopics` is the set of claims that the trusted issuer is allowed to emit
+     */
+    event TrustedIssuerAdded(IClaimIssuer indexed trustedIssuer, uint256[] claimTopics);
+
+    /**
+     *  this event is emitted when an issuer is removed from the trusted list.
+     *  the event is emitted by the removeTrustedIssuer function
+     *  `trustedIssuer` is the address of the trusted issuer's ClaimIssuer contract
+     */
+    event TrustedIssuerRemoved(IClaimIssuer indexed trustedIssuer);
+
+    /**
+     *  this event is emitted when the set of claim topics is changed for a given trusted issuer.
+     *  the event is emitted by the updateIssuerClaimTopics function
+     *  `trustedIssuer` is the address of the trusted issuer's ClaimIssuer contract
+     *  `claimTopics` is the set of claims that the trusted issuer is allowed to emit
+     */
+    event ClaimTopicsUpdated(IClaimIssuer indexed trustedIssuer, uint256[] claimTopics);
+
+    modifier onlyVerifiedSender() {
+        require(verify(_msgSender()), "sender is not verified");
```

Comment:
the Verifier contract should implement errors instead of require, the same way as it was implemented for the gateway contract. From now on we should use errors everywhere in smart contracts, as it is going to help understand why a transaction fails

Reply:
I keep the revert message of the current implementation of the TREX Identity Registry. If we use custom error, compatibility with existing packages and services that depends on these revert messages might break. Maybe we can keep the revert messages for the first implementation?